queue/queue_test.go

```diff
@@ -0,0 +1,46 @@
+package queue
```

Comment:
I'd like to have a test that runs all the Queue operations and with some mocked data if possible. 
Then we can run it with `-race` for some hours to be certain there are non data races.